src/v/ssx/watchdog.h

```diff
@@ -77,9 +77,18 @@ class watchdog {
         cancel();
     }
     watchdog(const watchdog&) = delete;
-    watchdog(watchdog&&) = delete;
+    watchdog(watchdog&& other) noexcept
+      : _as(std::move(other._as))
```

Comment:
prefer something like `_as(std::exchange(other._as, {}))` to make it clear that the other abort source should no longer be active.